### PR TITLE
Replaced Game Titles with their Icon

### PR DESCRIPTION
**Pull request by HenriTom**
Replaced the game icons with their icon resolving #148 
Added an option for that in the Misc tab

### CODE REVIEW

File: src/main/java/net/asodev/islandutils/util/IngameIcons.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        IslandUtilsEvents.GAME_CHANGE.register((game) -> {
+            if (game == Game.HUB || !IslandOptions.getMisc().useGameIcons())
+                return;
+
+            new Thread(() -> {
+                try {
+                    draw = true;
+                    Thread.sleep(5100);
+                    draw = false;
+                    MC.options.hideGui = false;
```
Comment (AsoDesu) [inline src/main/java/net/asodev/islandutils/util/IngameIcons.java:37]: please don't ever forcibly hide or unhide the player's gui.

File: src/main/java/net/asodev/islandutils/util/IngameIcons.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import net.asodev.islandutils.options.categories.MiscOptions;
+import net.asodev.islandutils.state.Game;
+import net.asodev.islandutils.state.MccIslandState;
+import net.fabricmc.fabric.api.client.rendering.v1.HudRenderCallback;
+import net.minecraft.client.DeltaTracker;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.resources.ResourceLocation;
+
+public class IngameIcons {
```
Comment (AsoDesu) [inline src/main/java/net/asodev/islandutils/util/IngameIcons.java:16]: this should be named `InGameIcons`

File: src/main/java/net/asodev/islandutils/util/IngameIcons.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+    public static void drawCenteredIcon(GuiGraphics guiGraphics, String name, int size) {
+        ResourceLocation texture = ResourceLocation.fromNamespaceAndPath("island", "textures/gui/games/" + name + ".png");
+
+        if (MC.screen != null || !isTexture(texture))
+            return;
+
+        if (draw) {
+            guiGraphics.blit(texture, (int) (MC.getWindow().getWidth() / MC.getWindow().getGuiScale() / 2 - size / 2), (int) (MC.getWindow().getHeight() / MC.getWindow().getGuiScale() / 2 - size / 2), 0, 0, size, size, size, size);
+            MC.options.hideGui = true;
```
Comment (AsoDesu) [inline src/main/java/net/asodev/islandutils/util/IngameIcons.java:53]: do not hide gui every frame.

**Review by AsoDesu — CHANGES_REQUESTED** [review]
this isn't really the effect that we're going for,

what we are trying to do is replicate the game titles at the start of games in MCC ([see this clip as an example, with the parkour warrior title](https://youtube.com/clip/Ugkxyy6hPC77UkMl9clZDyUCfmTvQnbGjoQk?si=wGWBnO5G7infGUhQ))

the icons displayed are too small, and need to be faded in and out properly
also please do not forcibly hide and unhide the players ui, it can and will get really annoying


my suggestion is instead of rendering the icons yourself, load in the game titles in a font, and then show them as a title (which is exactly what MCC does in the event)
you also may have to use a mixin to remove the current title which displays when you enter a game:

https://github.com/user-attachments/assets/78b0aaec-0902-4a30-98d9-0390f9cec41d

**Comment by UltraTeal** [comment]
So the logos aren’t the correct ones they are the old island icons the new ones are in the drive I sent tho the dynaball one is right

**Pull request closed without merging**